### PR TITLE
NEW: check changed state text fields on keyup, rather than just change

### PR DESCRIPTION
**Pull request by unclecheese**
This issue relates to the state of the save/publish buttons in CMSMain. They only update on blur of the text fields, creating an awkward UX, since many users will click one of those actions whilst the text fields are focused.

Minor: update changetracker to use on() and off() instead of bind() and unbind().

### CODE REVIEW

**Comment by jonom** [comment]
@unclecheese any interest (and available time) in picking this up again? Would love to see this niggling issue resolved.

**Comment by tractorcow** [comment]
Oh, this PR hasn't been touched for years.... sorry. :)

@unclecheese maybe you have the inclination? :D

**Comment by unclecheese** [comment]
It's probably not compatible with all the TinyMCE changes, is it?

**Comment by tractorcow** [comment]
I doubt that it is, honestly. :D

It's probably just worth checking whether the current master has this issue at all.

**Comment by jonom** [comment]
Would be great to get this fixed in 3 branch in any case... I imagine many people will still be using 3 even for new installs for a while after 4.0 ships as they'll wait for it to be proven to be very stable and module ecosystem to catch up, and when you consider upgrading of existing sites on the 3.x line a lot of people would probably benefit from this fix.

**Comment by jonom** [comment]
Looks like this isn't an issue in master for `HTMLEditorField` :smile: but it's still an issue for `TextField` and `TextareaField` :disappointed:

**Comment by unclecheese** [comment]
The issue is that paste doesn't work?

**Comment by jonom** [comment]
The issue is the original one you described (I didn't test paste)

> This issue relates to the state of the save/publish buttons in CMSMain. They only update on blur of the text fields, creating an awkward UX, since many users will click one of those actions whilst the text fields are focused.

**Comment by jonom** [comment]
This issue was really important to me as I assumed it was the root cause of https://github.com/silverstripe/silverstripe-framework/issues/5201 but looking at it a little today I'm not sure they're related. Would still love to see this issue fixed but #5201 seems more critical.

File: javascript/HtmlEditorField.js [hunk 1]
```diff
@@ -106,14 +106,27 @@ ss.editorWrappers.tinyMCE = (function() {
...
+					clearTimeout(timeoutID);
+					timeoutID = setTimeout(later, wait);
+				};
+			};	
 
+			// Update underlying textarea on every change, so external handlers
+			// such as changetracker have a chance to trigger properly.
+			this.instance.onKeyUp.add(changeHandler(250));			
+			
+			// Add more events here as needed.
```
Comment (jonom) [inline javascript/HtmlEditorField.js]: Suggest: onpropertychange, onchange, onkeyup, oninput, onpaste, oncut

**Comment by unclecheese** [comment]
Closing and resubmitting to the 3 branch.

**Pull request closed without merging**